vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -113,25 +101,35 @@ def _action_callback_function(self, **inputs: Dict[str, Any]) -> Dict[str, Any]:
         logger.debug(f"Action inputs: {inputs}")
 
         # Invoking the action's function
-        return_value = self.function(**inputs) or {}
+        return_value = self.function(**inputs)
 
         # Action callback outputs
         outputs = list(ctx.outputs_grouping.keys())
         outputs.remove("action_finished")
```

Comment:
There was a question: "Is there some way of doing this without access ctx? Since the calling function has access to callback_outputs already?"

There are few ways how we can make the `callback_outputs` variable visible in the `self._action_callback_function`:
1. propagate `callback_outputs` into `self._action_callback_function`,
2. store `callback_outputs` into `self.__callback_outputs`,
3. move `self._action_callback_function` body into the `callback_wrapper`.

1. Doesn't seem like the best approach since we need to reserve one word and forbid users to use it as an input.
2. This one looks okay to me (but it means we need to make significant changes in the many tests)
3. Makes the current `_action_callback_function` implementation impossible to test.

I'd appreciate your @antonymilne opinion here, maybe I missed something.